src/main/resources/changesets/changelog-consent-2025-05-01-add-era-commons-to-dar.xml

```diff
@@ -0,0 +1,18 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+  <changeSet id="changelog-consent-2025-05-01-add-era-commons-to-dar" author="grushton">
+    <addColumn tableName="data_access_request">
+      <column name="era_commons_id" type="varchar(255)"/>
+    </addColumn>
+    <sql>
+      UPDATE data_access_request dar
+      SET era_commons_id = users.era_commons_id FROM (select user_id, era_commons_id FROM users) AS users
+      WHERE dar.user_id = users.user_id
+      AND dar.submission_date IS NOT NULL;
```

Comment:
It would be nice if we could make this not null. We know that it won't be null in the future, correct?

Reply:
On dev, at least, I know there are null values. It is also a valid use case that a user can submit a dar, then remove their era commons link. It would be nice to enforce this more but I think we need to review the workflows in place.

Reply:
> then remove their era commons link

Do you mean remove it from the `User` or from the dar? If they remove it from `User` the dar isn't affected (in the future, once this code is in place) so the dar should still be non-null.

For the historical null value cases, we could put a placeholder value like `UNKNOWN`.

Reply:
> Do you mean remove it from the User or from the dar?

A user can unlink their account which removes it from their user, not anything else. 

> dar should still be non-null

Draft DARs will have this value populated until submission so it has to be nullable.

> placeholder value like `UNKNOWN`

We could do that. It's not required, but I'm also not sure what that gets us. The UI will know either way if the value is defined or `UNKNOWN`

Reply:
@rushtong  - your responses make sense to me given the state machine around the presence of an eRA Commons Id and the state machine around DataAccessRequest eRA Commons Id in this.  One thought about adding the "UNKNOWN" value that @pshapiro4broad raises is with this value set, we could consider adding eraCommonsId to our state machine check on DARs to know they're draft.

Reply:
If we add test coverage to that contract, we'll know when we've got code that is attempting to do something our expected state machine doesn't support.